### PR TITLE
[dagster-airlift][federation-apis] Airflow instance methods

### PR DESCRIPTION
**Pull request by dpeng817**
## Summary & Motivation
Publicize the instance methods that are useful for the federation use case and add docstrings.

## How I Tested These Changes
eyes

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25859?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25861** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25861?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25860** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25860?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25859** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25859?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25858** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25858?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25857** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25857?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25837** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25837?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25836** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25836?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25835** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25835?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25833** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25833?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25832** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25832?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25831** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25831?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25830** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25830?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25829** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25829?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25828** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25828?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25827** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25827?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25823** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25823?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25822** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25822?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25821** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25821?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25820** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25820?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25819** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25819?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25816** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25816?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25815** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25815?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25814** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25814?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py [hunk 1]
```diff
@@ -265,7 +266,19 @@ def get_dag_runs_batch(
...
 
+    @public
     def trigger_dag(self, dag_id: str, logical_date: Optional[datetime.datetime] = None) -> str:
+        """Trigger a dag run for the given dag_id.
+
+        Does not wait for the run to finish. To wait for the completed run to finish, use :py:meth:`wait_for_run_completion`.
+
+        Args:
+            dag_id (str): The dag id to trigger.
+            logical_date (Optional[datetime.datetime]): The logical date to use for the dag run. If not provided, the current time will be used.
```
Comment (benpankow) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:277]: what's a logical date? (if this is a term airflow users will be familiar with, prob no need to add, but as a layperson it is not clear what this indicates)

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py [hunk 2]
```diff
@@ -314,7 +338,17 @@ def wait_for_run_completion(self, dag_id: str, run_id: str, timeout: int = 30) -
...
+    @public
     def get_run_state(self, dag_id: str, run_id: str) -> str:
+        """Given a run ID of an airflow dag, return the state of that run.
+
+        Args:
+            dag_id (str): The dag id.
+            run_id (str): The run id.
+
+        Returns:
+            str: The state of the run. Will be one of the states defined by Airflow.
```
Comment (benpankow) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:350]: maybe doc link to list of states?

**Review by benpankow — APPROVED** [review]

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:277]: Yea this is an airflow-specific term.

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py (hunk 2 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:350]: I tried to find it; it doesn't exist 😭

**Pull request merged**